### PR TITLE
MRG: Add fnirs support

### PR DESCRIPTION
**Pull request by rob-luke**
Thanks for the great package.

I have been using this small modification to use auto reject with my data. Are you interested in adding fNIRS support to this package? If so, I can expand this PR where necessary.

#### Notes

I will squash commits before asking for a merge.

### CODE REVIEW

**Comment by agramfort** [comment]
I need no objection. Does it work with your data?

**Comment by rob-luke** [comment]
> Does it work with your data?

Absolutely! It should be called magicreject.

Ok, I will look through the code a bit more and make some changes. I will ping @agramfort when I think its ready for a review.

**Comment by agramfort** [comment]
really great to hear !

>

**Comment by jasmainak** [comment]
Love it, let us know when ready!

**Comment by jasmainak** [comment]
@rob-luke do you want to rebase this PR?

**Comment by rob-luke** [comment]
Thanks @jasmainak, I will give this a good test against a few standard datasets and ping you in coming days once I'm happy. Time to get this merged I think 😄

**Comment by jasmainak** [comment]
perfect, will do!

**Comment by rob-luke** [comment]
@jasmainak I added some tests, but the CIs won't run without permission. Could you please hit the run button. Thanks

File: autoreject/tests/test_autoreject.py [hunk 1]
```diff
@@ -318,3 +319,33 @@ def test_io():
     assert_array_equal(epochs_clean1.get_data(), epochs_clean3.get_data())
     assert_array_equal(reject_log1.labels, reject_log2.labels)
     assert_array_equal(reject_log1.labels, reject_log3.labels)
+
+
+def test_fnirs():
+    # Test that autoreject runs on fNIRS data.
+    raw = mne.io.read_raw_nirx(os.path.join(
```
Comment (rob-luke) [inline autoreject/tests/test_autoreject.py:326]: Is this too much? We can remove if you think its over the top.

Comment (jasmainak) [reply autoreject/tests/test_autoreject.py:326]: how big is the dataset?

File: autoreject/tests/test_autoreject.py [hunk 2]
```diff
@@ -318,3 +319,33 @@ def test_io():
...
+                                                           '3.0': 3})
+    event_dict = {'Control': 1, 'Tapping/Left': 2, 'Tapping/Right': 3}
+    epochs = mne.Epochs(raw, events, event_id=event_dict,
+                        tmin=-5, tmax=15,
+                        proj=True, baseline=(None, 0), preload=True,
+                        detrend=None, verbose=True)
+    # Test autoreject
+    ar = AutoReject()
+    assert len(epochs) == 37
+    epochs_clean = ar.fit_transform(epochs)
```
Comment (jasmainak) [inline autoreject/tests/test_autoreject.py:342]: sorry for my ignorance, how does interpolation work for this kind of data? Are there many channels?

**Comment by jasmainak** [comment]
@rob-luke it looks like you need to rebase

**Comment by rob-luke** [comment]
> @rob-luke it looks like you need to rebase

Done (I think).

File: autoreject/tests/test_autoreject.py (hunk 1 above)
Comment (rob-luke) [reply autoreject/tests/test_autoreject.py:326]: A total size of 74 MB

File: autoreject/tests/test_autoreject.py (hunk 2 above)
Comment (rob-luke) [reply autoreject/tests/test_autoreject.py:342]: Currently in MNE we are replacing the data with the closest channel. Until I get around to developing something smarter, but that's not on my schedule for a few months. So maybe we should remove this?

The number of channels varies greatly. There are single channel systems through to high density fNIRS which probably has 100s of channels. The high density systems are commonly available though. For reference, my two labs typically use around 50 channels.

Comment (rob-luke) [reply autoreject/tests/test_autoreject.py:342]: Here is a figure of what a low (left) and high density (right) system would look like targeting the frontal cortex. This is from a commercial system.

green lines represent channels

image

File: autoreject/tests/test_autoreject.py (hunk 1 above)
Comment (jasmainak) [reply autoreject/tests/test_autoreject.py:326]: okay that's fine I think

File: autoreject/tests/test_autoreject.py [hunk 3]
```diff
@@ -318,3 +319,33 @@ def test_io():
     assert_array_equal(epochs_clean1.get_data(), epochs_clean3.get_data())
     assert_array_equal(reject_log1.labels, reject_log2.labels)
     assert_array_equal(reject_log1.labels, reject_log3.labels)
+
+
+def test_fnirs():
+    # Test that autoreject runs on fNIRS data.
```
Comment (jasmainak) [inline autoreject/tests/test_autoreject.py:325]:
```suggestion
    """Test that autoreject runs on fNIRS data."""
```

Just a proper docstring so pytest can use it

File: autoreject/tests/test_autoreject.py (hunk 2 above)
Comment (jasmainak) [reply autoreject/tests/test_autoreject.py:342]: I see, got it. It's fine as such I think. Feel free to make follow up pull requests to improve the functionality. I'm guessing you look for the nearest channel of the same channel type?

**Comment by jasmainak** [comment]
all looks good @rob-luke, do you want to update `whats_new.rst` ?

File: autoreject/tests/test_autoreject.py (hunk 2 above)
Comment (rob-luke) [reply autoreject/tests/test_autoreject.py:342]: > I'm guessing you look for the nearest channel of the same channel type?

Correct

File: autoreject/autoreject.py [hunk 4]
```diff
@@ -185,9 +185,9 @@ def get_rejection_threshold(epochs, decim=1, random_state=None,
     ch_types : str | list of str | None
         The channel types for which to find the rejection dictionary.
         e.g., ['mag', 'grad']. If None, the rejection dictionary
-        will have keys ['mag', 'grad', 'eeg', 'eog'].
-    cv : a scikit-learn cross-validation object
-        Defaults to cv=5
+        will have keys ['mag', 'grad', 'eeg', 'eog', 'hbo', 'hbr'].
+    cv : int
+        The number of folds used. Defaults to 5.
```
Comment (sappelhoff) [inline autoreject/autoreject.py:190]: Why this change?

Comment (rob-luke) [reply autoreject/autoreject.py]: Thanks for catching that. That was a rebase error by me.

**Comment by jasmainak** [comment]
Thanks @rob-luke for this fantastic contribution and thanks @sappelhoff for the review (sharp eye!)
